File: backend/verify_api.py
import requests
import os

url = 'http://127.0.0.1:10000/api/convert/ncn-to-dxf'
test_file_path = 'test_sample.ncn'

# Create a sample NCN file
with open(test_file_path, 'w') as f:
    f.write('1 500000 4500000 100\n')
    f.write('2 500010 4500010 102.5\n')

try:
    with open(test_file_path, 'rb') as f:
        r = requests.post(url, files={'file': f})
    
    if r.status_code == 200:
        print(f"Success! Status: {r.status_code}")
        print(f"Downloaded DXF size: {len(r.content)} bytes")
        with open('output_test.dxf', 'wb') as f:
            f.write(r.content)
        print("Output saved to output_test.dxf")
    else:
        print(f"Error! Status: {r.status_code}")
        print(f"Response: {r.text}")
finally:
    if os.path.exists(test_file_path):
        os.remove(test_file_path)

File: backend/verify_api.js
const axios = require('axios');
const fs = require('fs');
const FormData = require('form-data');

async function testApi() {
    const url = 'http://127.0.0.1:10000/api/convert/ncn-to-dxf';
    const ncnContent = '1 500000 4500000 100\n2 500010 4500010 102.5';
    fs.writeFileSync('test_sample.ncn', ncnContent);

    const form = new FormData();
    form.append('file', fs.createReadStream('test_sample.ncn'));

    try {
        console.log('Sending request to:', url);
        const response = await axios.post(url, form, {
            headers: form.getHeaders(),
            responseType: 'arraybuffer'
        });

        console.log('Success! Status:', response.status);
        console.log('Downloaded DXF size:', response.data.length, 'bytes');
        fs.writeFileSync('output_test.dxf', response.data);
        console.log('Output saved to output_test.dxf');
    } catch (error) {
        console.error('Error! Status:', error.response ? error.response.status : 'N/A');
        console.error('Message:', error.message);
        if (error.response && error.response.data) {
            console.error('Response:', Buffer.from(error.response.data).toString());
        }
    } finally {
        if (fs.existsSync('test_sample.ncn')) fs.unlinkSync('test_sample.ncn');
    }
}

testApi();
